packages/create-block/lib/templates.js

```diff
@@ -65,27 +72,42 @@ const templates = {
 	},
 };
 
-const getTemplate = ( templateName ) => {
-	if ( ! templates[ templateName ] ) {
+const getTemplate = async ( templateName ) => {
+
+	const isCoreTemplate = await checkIsCoreTemplate( templateName );
+	if ( isCoreTemplate ) {
+		return templates[ templateName ];
+	}
+
+	// throw a CLIError if the the template is neither a core one nor an external one
+	if ( ! await checkIsExternalTemplate( templateName ) && ! isCoreTemplate ) {
 		throw new CLIError(
-			`Invalid template type name. Allowed values: ${ Object.keys(
+			`Invalid template type name. Either use one of the Core templates: ${ Object.keys(
 				templates
-			).join( ', ' ) }.`
+			).join( ', ' ) }. \n \n or a valid npm package name.`
 		);
 	}
-	return templates[ templateName ];
+
+	await downloadExternalTemplate( templateName );
+
+	const rawPackageInfo = await readFile( join( process.cwd(), 'temp', 'node_modules', templateName, 'template.json' ) );
+	const packageInfo = JSON.parse( rawPackageInfo );
+	return packageInfo;
```

Comment:
How about you move it to `downloadExternalTemplate`?

Reply:
Moved  👍